### PR TITLE
pdf-viewer: Rename deprecated methods

### PR DESCRIPTION
**Pull request by c1-g**
Rename all PyMuPDF's methods and properties in pdf-viewer python code according to https://pymupdf.readthedocs.io/en/latest/znames.html since it will remove them soon.

### CODE REVIEW

**Comment by manateelazycat** [comment]
@c1-g Did you test your patch before commit ?

File: app/pdf-viewer/buffer.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def set_focus_text(self, new_text):
 
     def get_toc(self):
         result = ""
-        toc = self.buffer_widget.document.getToC()
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:206]: fitz.Document, this library use getToC() to get table, did you know this is library API?

File: app/pdf-viewer/buffer.py [hunk 2]
```diff
@@ -249,10 +249,10 @@ def __getitem__(self, index):
             if not self._is_trim_margin:
                 return page
 
-            if page.CropBox == self._document_page_clip:
+            if page.cropbox == self._document_page_clip:
                 return page
 
-        page = PdfPage(self.document[index], self.document.isPDF)
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:255]: isPDF is fitz.Document's API, did you know?

**Comment by MatthewZMD** [comment]
This patch is completely broken, it doesn't even start. 

Please test your code before submitting a PR, to see at least it *can* run. This is not the first time we're telling you this, it is a very simple mistake that costs us time and patience.

File: app/pdf-viewer/buffer.py (hunk 1 above)
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:206]: fitz.Document, this library use getToC() to get table, did you know this is library API?

File: app/pdf-viewer/buffer.py [hunk 3]
```diff
@@ -326,15 +326,15 @@ def get_page_width(self):
         if self.isPDF:
             if self._is_trim_margin:
                 return self._document_page_clip.width
-            return self.document.pageCropBox(0).width
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:329]: pageCropBox is fitz.Document's API!

File: app/pdf-viewer/buffer.py [hunk 4]
```diff
@@ -359,18 +359,18 @@ def __getattr__(self, attr):
         return getattr(self.page, attr)
 
     def _init_page_rawdict(self):
-        if self.isPDF:
+        if self.isPDFr:
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:362]: Did you test this typo?

File: app/pdf-viewer/buffer.py [hunk 5]
```diff
@@ -359,18 +359,18 @@ def __getattr__(self, attr):
...
             # Must set CropBox before get page rawdict , if no,
             # the rawdict bbox coordinate is wrong
             # cause the select text failed
-            self.page.setCropBox(self.clip)
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:366]: setCropBox is fitz.Page API!

File: app/pdf-viewer/buffer.py [hunk 6]
```diff
@@ -359,18 +359,18 @@ def __getattr__(self, attr):
...
-            d = self.page.getText("rawdict")
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:367]: getText is API

File: app/pdf-viewer/buffer.py [hunk 7]
```diff
@@ -359,18 +359,18 @@ def __getattr__(self, attr):
...
+            self.page.set_cropbox(self.clip)
+            d = self.page.get_text("rawdict")
             # cancel the cropbox, if not, will cause the pixmap set cropbox
             # don't begin on top-left(0, 0), page display black margin
-            self.page.setCropBox(self.page.MediaBox)
+            self.page.set_cropbox(self.page.mediabox)
             return d
         else:
-            return self.page.getText("rawdict")
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:373]: getText is API

File: app/pdf-viewer/buffer.py [hunk 8]
```diff
@@ -409,13 +409,13 @@ def _init_tight_margin(self):
...
             r = fitz.Rect(x0, y0, x1, y1)
         if r is None:
-            return self.page.CropBox
+            return self.page.cropbox
         return r
 
     def get_tight_margin_rect(self):
         # if current page don't computer tight rect
         # return None
-        if self._tight_margin_rect == self.page.MediaBox:
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:418]: MediaBox is API

File: app/pdf-viewer/buffer.py [hunk 9]
```diff
@@ -435,18 +435,18 @@ def get_page_char_rect_index(self, x, y):
...
         else:
-            self.page_width = self.page.CropBox.width
-            self.page_height = self.page.CropBox.height
+            self.page_width = self.page.cropbox.width
+            self.page_height = self.page.cropbox.height
 
     def get_qpixmap(self, scale, *args):
         if self.isPDF:
-            self.page.setCropBox(self.clip)
-        pixmap = self.page.getPixmap(matrix=fitz.Matrix(scale, scale), alpha=False)
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:449]: getPixmap is API

File: app/pdf-viewer/buffer.py [hunk 10]
```diff
@@ -462,73 +462,73 @@ def with_invert(self, invert, exclude_image=True):
         # if exclude image is True, will find the page all image, then get
         # each image rect. Finally, again invert all image rect.
         def fn(page, pixmap, scale):
-            pixmap.invertIRect(pixmap.irect)
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:465]: invertIRect is API

**Comment by manateelazycat** [comment]
Please don't rename those functions, pymupdf said old name still can use.

We need make sure almost user switch to new version of PyMuPDF, then refactory code.

This patch break all EAF users.

File: app/pdf-viewer/buffer.py [hunk 11]
```diff
@@ -462,73 +462,73 @@ def with_invert(self, invert, exclude_image=True):
...
+                self.page.delete_annot(annot)
             self._mark_link_annot_list = []
 
     def mark_search_text(self, keyword):
-        quads_list = self.page.searchFor(keyword, hit_max=999, quads=True)
+        quads_list = self.page.search_for(keyword, hit_max=999, quads=True)
         if quads_list:
             for quads in quads_list:
-                annot = self.page.addHighlightAnnot(quads)
+                annot = self.page.add_highlight_annot(quads)
```
Comment (manateelazycat) [inline app/pdf-viewer/buffer.py:510]: add_highlight_annot is not exist in my Pymupdf version.

I don't know other changed functions is exist?

**Comment by manateelazycat** [comment]
Please don't send any patch that pass test, include typo to break everybody.

**Pull request merged**